wrapanapi/systems/msazure.py

```diff
@@ -4,52 +4,444 @@
+        }
+        public_ip = self.system.network_client.public_ip_addresses.create_or_update(
+            resource_group_name=resource_group,
+            public_ip_address_name=vm_name,
+            parameters=public_ip_params
+        ).result()
 
+        # creating virtual network
+        virtual_networks = self.system.network_client.virtual_networks
+        if vnet_name not in [v.name for v in virtual_networks.list(resource_group)]:
+            vnet_params = {
+                'location': location,
+                'address_space': {
+                    'address_prefixes': [address_space]
+                }
+            }
+            virtual_networks.create_or_update(
+                resource_group_name=resource_group,
+                virtual_network_name=vnet_name,
+                parameters=vnet_params
+            ).result()
 
-# TODO: add handler for logger msrest.service_client if needed
-# TODO: add better description to each method
+        # creating sub net
+        subnet_name = 'default'
+        subnets = self.system.network_client.subnets
+        if subnet_name not in [v.name for v in subnets.list(resource_group, vnet_name)]:
+            vsubnet = subnets.create_or_update(
+                resource_group_name=resource_group,
+                virtual_network_name=vnet_name,
+                subnet_name='default',
+                subnet_parameters={'address_prefix': subnet}
+            ).result()
+        else:
+            vsubnet = subnets.get(
+                resource_group_name=resource_group,
+                virtual_network_name=vnet_name,
+                subnet_name='default')
 
+        # creating network interface
+        nic_params = {
+            'location': location,
+            'ip_configurations': [{
+                'name': vm_name,
+                'public_ip_address': public_ip,
+                'subnet': {
+                    'id': vsubnet.id
+                }
+            }]
+        }
+        nic = self.system.network_client.network_interfaces.create_or_update(
+            resource_group_name=resource_group,
+            network_interface_name=vm_name,
+            parameters=nic_params
+        ).result()
 
-class AzureSystem(WrapanapiAPIBaseVM):
+        # preparing os disk
+        # todo: replace with copy disk operation
+        self.system.copy_blob_image(
+            self.name, vm_name, vm_settings['storage_account'],
+            self._container, storage_container
+        )
+        image_uri = self.system.container_client.make_blob_url(
+            container_name=storage_container, blob_name=vm_name)
+        # creating virtual machine
+        vm_parameters = {
+            'location': location,
+            'hardware_profile': {
+                'vm_size': getattr(VirtualMachineSizeTypes, vm_size)
+            },
+            'storage_profile': {
+                'os_disk': {
+                    'os_type': 'Linux',  # TODO: why is this hardcoded?
```

Comment:
we should switch to creating VMs from Images by default. this code is outdated - it was creating VM from the existing blod - vhd disk. but again - not in this PR